In [1]:
import grpc

import server_pb2
import server_pb2_grpc

In [2]:
# Create a channel
channel = grpc.insecure_channel('andromeda.fi.muni.cz:58110')

In [3]:
# Create a stub (client)
stub = server_pb2_grpc.HomeworkStub(channel)

In [4]:
# Create a request with university email
request = server_pb2.BeginData(email='555179@mail.muni.cz')

In [5]:
# Make the call to the Begin method
response = stub.Begin(request)
session_token = response.token

In [6]:
response

token: "b4491ecf-806b-4dea-a09c-69a854c386a6"
info: "Define the following two messages:\n    message AdderTask {\n        int64 a = 1;\n        float b = 2;\n    }\n\n    message AdderTaskResponse {\n        string token = 1;\n        double result = 2;\n    }\n\nThe AdderTask in \"oneof tasks\" in the Response message has the field number 8.\n\nExtend the Homework service with the following method:\n    rpc Adder(AdderTaskResponse) returns (Response);\n\nTo get the next task, you have to call the Adder service with the response filled\nwith your assigned token and the result of the sum of the values \"a\" and \"b\" in\nthe AdderTask.\n"
adderTask {
  a: 610
  b: 107.040146
}

info: "Define the following two messages:

message AdderTask {

    int64 a = 1;
    float b = 2;

}

message AdderTaskResponse {

    string token = 1;
    double result = 2;

}

The AdderTask in \"oneof tasks\" in the Response message has the field number 8.

Extend the Homework service with the following method:

rpc Adder(AdderTaskResponse) returns (Response);

To get the next task, you have to call the Adder service with the response filled with your assigned token and the result of the sum of the values \"a\" and \"b\" in the AdderTask."

adderTask {

    a: 610
    b: 107.040146

}



In [14]:
# Calculate the sum of a and b
a = response.adderTask.a
b = response.adderTask.b
result = a + b
result

717.0401458740234

In [15]:
adder_response = server_pb2.AdderTaskResponse(
    token=session_token,
    result=result
)


In [16]:
adder_response

token: "b4491ecf-806b-4dea-a09c-69a854c386a6"
result: 717.04014587402344

In [17]:
request2 = server_pb2.AdderTaskResponse(token=adder_response.token, result=adder_response.result)
response2 = stub.Adder(request2)

In [18]:
response2

token: "b4491ecf-806b-4dea-a09c-69a854c386a6"
info: "Define the following two messages:\n    message MatrixTask {\n        message Row {\n            repeated int64 values = 1; \n        }\n        repeated Row rows = 1;\n    }\n\n    message MatrixTaskResponse {\n        string token = 1;\n        int64 determinant = 2;\n    }\n\nThe MatrixTask in \"oneof tasks\" in the Response message has the field number 9.\n\nExtend the Homework service with the following method:\n    rpc Matrix(MatrixTaskResponse) returns (Response);\n\nTo get the next task, you have to call the Matrix method with the response filled\nwith your assigned token and the matrix determinant.\nSee https://en.wikipedia.org/wiki/Determinant for the description of the matrix determinant."